<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Изучение-данных" data-toc-modified-id="Изучение-данных-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Изучение данных</a></span></li><li><span><a href="#Предобработка-данных" data-toc-modified-id="Предобработка-данных-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Предобработка данных</a></span><ul class="toc-item"><li><span><a href="#Обработка-пропусков" data-toc-modified-id="Обработка-пропусков-2.1"><span class="toc-item-num">2.1&nbsp;&nbsp;</span>Обработка пропусков</a></span></li><li><span><a href="#Замена-типа-данных" data-toc-modified-id="Замена-типа-данных-2.2"><span class="toc-item-num">2.2&nbsp;&nbsp;</span>Замена типа данных</a></span></li><li><span><a href="#Обработка-дубликатов" data-toc-modified-id="Обработка-дубликатов-2.3"><span class="toc-item-num">2.3&nbsp;&nbsp;</span>Обработка дубликатов</a></span></li><li><span><a href="#Лемматизация" data-toc-modified-id="Лемматизация-2.4"><span class="toc-item-num">2.4&nbsp;&nbsp;</span>Лемматизация</a></span></li><li><span><a href="#Категоризация-данных" data-toc-modified-id="Категоризация-данных-2.5"><span class="toc-item-num">2.5&nbsp;&nbsp;</span>Категоризация данных</a></span></li></ul></li><li><span><a href="#Результаты-анализа" data-toc-modified-id="Результаты-анализа-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Результаты анализа</a></span><ul class="toc-item"><li><span><a href="#Есть-ли-зависимость-между-наличием-детей-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-наличием-детей-и-возвратом-кредита-в-срок?-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Есть ли зависимость между наличием детей и возвратом кредита в срок?</a></span></li><li><span><a href="#Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-семейным-положением-и-возвратом-кредита-в-срок?-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Есть ли зависимость между семейным положением и возвратом кредита в срок?</a></span></li><li><span><a href="#Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?" data-toc-modified-id="Есть-ли-зависимость-между-уровнем-дохода-и-возвратом-кредита-в-срок?-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Есть ли зависимость между уровнем дохода и возвратом кредита в срок?</a></span></li><li><span><a href="#Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?" data-toc-modified-id="Как-разные-цели-кредита-влияют-на-его-возврат-в-срок?-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Как разные цели кредита влияют на его возврат в срок?</a></span></li></ul></li><li><span><a href="#Вывод" data-toc-modified-id="Вывод-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Вывод</a></span></li></ul></div>

In [1]:
#%pip install --upgrade pip
#%pip install -q -r requirements.txt

# Исследование надёжности заёмщиков

## Изучение данных

Для начала откроем файл, выведем первые 10 строк и получим общую информацию о таблице:

In [2]:
# импорт библиотеки pandas
import pandas as pd

# чтение файла с данными и сохранение в переменную data
data = pd.read_csv('/datasets/data.csv')

In [3]:
# вывод первых 10 строк
data.head(10)

children  days_employed  dob_years education  education_id  \
0         1   -8437.673028         42    высшее             0   
1         1   -4024.803754         36   среднее             1   
2         0   -5623.422610         33   Среднее             1   
3         3   -4124.747207         32   среднее             1   
4         0  340266.072047         53   среднее             1   
5         0    -926.185831         27    высшее             0   
6         0   -2879.202052         43    высшее             0   
7         0    -152.779569         50   СРЕДНЕЕ             1   
8         2   -6929.865299         35    ВЫСШЕЕ             0   
9         0   -2188.756445         41   среднее             1   

      family_status  family_status_id gender income_type  debt   total_income  \
0   женат / замужем                 0      F   сотрудник     0  253875.639453   
1   женат / замужем                 0      F   сотрудник     0  112080.014102   
2   женат / замужем                 0      M   сотрудник     0  145885.952297   
3   женат / замужем                 0      M   сотрудник     0  267628.550329   
4  гражданский брак                 1      F   пенсионер     0  158616.077870   
5  гражданский брак                 1      M   компаньон     0  255763.565419   
6   женат / замужем                 0      F   компаньон     0  240525.971920   
7   женат / замужем                 0      M   сотрудник     0  135823.934197   
8  гражданский брак                 1      F   сотрудник     0   95856.832424   
9   женат / замужем                 0      M   сотрудник     0  144425.938277   

                      purpose  
0               покупка жилья  
1     приобретение автомобиля  
2               покупка жилья  
3  дополнительное образование  
4             сыграть свадьбу  
5               покупка жилья  
6           операции с жильем  
7                 образование  
8       на проведение свадьбы  
9     покупка жилья для семьи

Получим общую информацию о таблице:

In [5]:
# получение общей информации о данных
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
children            21525 non-null int64
days_employed       19351 non-null float64
dob_years           21525 non-null int64
education           21525 non-null object
education_id        21525 non-null int64
family_status       21525 non-null object
family_status_id    21525 non-null int64
gender              21525 non-null object
income_type         21525 non-null object
debt                21525 non-null int64
total_income        19351 non-null float64
purpose             21525 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 2.0+ MB


В таблице 12 столбцов. В таблице разные типы данных - два float64, пять int64 и пять object.

Колонки содержат следующие данные:
1. children — количество детей в семье
2. days_employed — трудовой стаж в днях
3. dob_days — возраст клиента в годах
4. education — образование клиента
5. education_id — идентификатор образования
6. family_status — семейное положение
7. family_status_id — идентификатор семейного положения
8. gender — пол клиента
9. income_type — тип занятости
10. debt — имел ли задолженность по возврату кредитов
11. total_income — доход в месяц
12. purpose — цель получения кредита

**Вывод**

В каждой строке таблицы — данные о платежеспособности клиентов банка. Вероятно, данных достаточно для проверки влияния семейного положение и количества детей клиента на факт погашения кредита в срок. Но встречаются пропуски в данных в столбцах "трудовой стаж" и "доход в месяц".

## Предобработка данных

### Обработка пропусков

При первичном осмотре данных, видно, что значения в столбце "трудовой стаж" либо отрицательные, либо более 300 000. Скорее всего, в данных встречается информация из двух систем. Одна из них считает стаж в часах и "наоборот". Сделаем отрицательные значения положительные, а большие значения (более 300 000) переведем в дни (поделим на 24):

In [6]:
# преобразование отрицательных значений в столбце days_employed на положительные
data['days_employed'] = data['days_employed'].apply(abs)
# перевод больших значений столбца days_employed в дни
data.loc[data['days_employed'] > 300000, 'days_employed'] = data.loc[data['days_employed'] > 300000, 'days_employed'] / 24

Посчитаем количество пропусков в столбцах "трудовой стаж" и "доход в месяц".

In [7]:
# подсчет количества пропусов столбцов days_employed и total_income
print('Пропусков "трудового стажа":', data['days_employed'].isna().sum())
print('Пропусков "дохода в месяц":', data['total_income'].isna().sum())

Пропусков "трудового стажа": 2174
Пропусков "дохода в месяц": 2174


Трудовой стаж и доход в месяц пропущенны в 2174 строках. Это 2174/21525=10,1%. Удалять данные нельзя, так как они могут повлиять на выводы исследования. 

Выведем первые 10 пропущенных значений из столбца "трудовой стаж":

In [8]:
# вывод 10 строк из с пропущенными значениями days_employed
data[data['days_employed'].isna()].head(10)

children  days_employed  dob_years education  education_id  \
12         0            NaN         65   среднее             1   
26         0            NaN         41   среднее             1   
29         0            NaN         63   среднее             1   
41         0            NaN         50   среднее             1   
55         0            NaN         54   среднее             1   
65         0            NaN         21   среднее             1   
67         0            NaN         52    высшее             0   
72         1            NaN         32    высшее             0   
82         2            NaN         50    высшее             0   
83         0            NaN         52   среднее             1   

            family_status  family_status_id gender  income_type  debt  \
12       гражданский брак                 1      M    пенсионер     0   
26        женат / замужем                 0      M  госслужащий     0   
29  Не женат / не замужем                 4      F    пенсионер     0   
41        женат / замужем                 0      F  госслужащий     0   
55       гражданский брак                 1      F    пенсионер     1   
65  Не женат / не замужем                 4      M    компаньон     0   
67        женат / замужем                 0      F    пенсионер     0   
72        женат / замужем                 0      M  госслужащий     0   
82        женат / замужем                 0      F    сотрудник     0   
83        женат / замужем                 0      M    сотрудник     0   

    total_income                                purpose  
12           NaN                        сыграть свадьбу  
26           NaN                            образование  
29           NaN       строительство жилой недвижимости  
41           NaN       сделка с подержанным автомобилем  
55           NaN                        сыграть свадьбу  
65           NaN  операции с коммерческой недвижимостью  
67           NaN                покупка жилья для семьи  
72           NaN  операции с коммерческой недвижимостью  
82           NaN                                  жилье  
83           NaN                                  жилье

Видим, что значения NaN в колонках days_employed и total_income совпадают. Если удалить эти строки из-за пропусков, потеряем важные данные.
Чтобы не лишиться строк с важными данными, заполним значения NaN в указанных столбцах медианными значениями:

**Вывод**

In [9]:
income_median = data.groupby('income_type')['total_income'].median()
days_employed_median = data.groupby('income_type')['days_employed'].median()
print(income_median, days_employed_median)
for income_type in data['income_type'].unique():
    income_median = data.loc[data['income_type'] == income_type, 'total_income'].median()
    days_employed_median = data.loc[data['income_type'] == income_type, 'days_employed'].median()
    data.loc[(data['total_income'].isna()) & (data['income_type'] == income_type), 'total_income'] = income_median
    data.loc[(data['days_employed'].isna()) & (data['income_type'] == income_type), 'days_employed'] = days_employed_median

income_type
безработный        131339.751676
в декрете           53829.130729
госслужащий        150447.935283
компаньон          172357.950966
пенсионер          118514.486412
предприниматель    499163.144947
сотрудник          142594.396847
студент             98201.625314
Name: total_income, dtype: float64 income_type
безработный        15267.235531
в декрете           3296.759962
госслужащий         2689.368353
компаньон           1547.382223
пенсионер          15217.221094
предприниматель      520.848083
сотрудник           1574.202821
студент              578.751554
Name: days_employed, dtype: float64


Пропущенные значения были в двух столбцах, строки совпадали, то есть, в строке, где отсутствовал трудовой стаж, отсутствовал и доход в месяц. Для чистоты исследования, было решено заменить их на медианные значения. NaN - это специальные значения, которые указывают на пропуск. Таким образом, эти значения клиенты не указали намеренно, либо забыли.

### Замена типа данных

Заменим значения с вещественных на целочисленный, для удобства и производительности. Проверим полученные данные:

In [11]:
# замена типа данных на int в столбцах days_employed и total_income
data['days_employed'] = data['days_employed'].astype('int')
data['total_income'] = data['total_income'].astype('int')
# получение общей информации о данных в таблице data
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21525 entries, 0 to 21524
Data columns (total 12 columns):
children            21525 non-null int64
days_employed       21525 non-null int64
dob_years           21525 non-null int64
education           21525 non-null object
education_id        21525 non-null int64
family_status       21525 non-null object
family_status_id    21525 non-null int64
gender              21525 non-null object
income_type         21525 non-null object
debt                21525 non-null int64
total_income        21525 non-null int64
purpose             21525 non-null object
dtypes: int64(7), object(5)
memory usage: 2.0+ MB


**Вывод**

На данном этапе были изменены типы данных на целочисленные. Использовали метод astype(), он подходит для смены значений с вещественных на целочисленные.

### Обработка дубликатов

Проверим, есть ли в данных явные дубликаты, которые совпадают в срезе строки:

In [12]:
# получение информации о количестве явных дубликатов
data.duplicated().sum()

54

Явные дубликаты присутствуют и подсчитаны. Удалим их:

In [13]:
# удаление явных дубликатов (с удалением старых индексов и формированием новых)
data = data.drop_duplicates().reset_index(drop=True)

Проверим остались ли явные дубликаты:

In [14]:
# получение информации о количестве явных дубликатов
data.duplicated().sum()

0

При просмотре данных, видно, что в колонке "образование" встречается разный регистр. Проверим, какие значение и сколько упоминаний каждого:

In [15]:
# получение уникальных значений и их повторений в столбце education
data['education'].value_counts()

среднее                13705
высшее                  4710
СРЕДНЕЕ                  772
Среднее                  711
неоконченное высшее      668
ВЫСШЕЕ                   273
Высшее                   268
начальное                250
Неоконченное высшее       47
НЕОКОНЧЕННОЕ ВЫСШЕЕ       29
НАЧАЛЬНОЕ                 17
Начальное                 15
ученая степень             4
Ученая степень             1
УЧЕНАЯ СТЕПЕНЬ             1
Name: education, dtype: int64

Изменим регистр всех значений в колонке "образование" на нижний и повторно проверим их количество:

In [16]:
# изменение регистра в колонке education на нижний 
data['education'] = data['education'].str.lower() 
# получение уникальных значений и их повторений в столбце education
data['education'].value_counts()

среднее                15188
высшее                  5251
неоконченное высшее      744
начальное                282
ученая степень             6
Name: education, dtype: int64

Хоть в исходных данных и 15 вариантов, но на самом деле оказалось разных всего 5. После этого, проверим появились ли явные дубликаты:

In [17]:
# получение информации о количестве явных дубликатов
data.duplicated().sum() 

17

Вот и появились еще явные дубликаты, которые могли повлиять на результаты исследования. Удалим их и проверим количество:

In [18]:
# удаление явных дубликатов (с удалением старых индексов и формированием новых)
data = data.drop_duplicates().reset_index(drop=True)
# получение информации о количестве явных дубликатов
data.duplicated().sum() 

0

**Вывод**

На текущем этапе были удалены явные дубликаты. Для этого использовали метод drop_duplicates(). Регистр был нарушен только в одном столбце, после его изменения были повторно удалены явные дубликаты. Для предоставленных данных можем удалить только явные дубликаты, так как не хватает данных для выявления неявных дубликатов (например: id или ФИО с датой рождения клиента). Вероятней всего, дубликаты появились из-за повторной отправки заявки клиентом.

### Лемматизация

Чтобы выделить и сгруппировать данные по цели кредита, проведем лемматизацию столбца purpose:

In [19]:
# импорт библиотеки pymystem3
from pymystem3 import Mystem
m = Mystem()
# лемматизация уникальных значений в столбце purpose
text = data['purpose']
lemmas = m.lemmatize(' '.join(text))
# импорт библиотеки collections и подсчет частоты появления уникальных слов в столбце purpose
from collections import Counter
print(Counter(lemmas))

Counter({' ': 55023, 'недвижимость': 6351, 'покупка': 5897, 'жилье': 4460, 'автомобиль': 4306, 'образование': 4013, 'с': 2918, 'операция': 2604, 'свадьба': 2324, 'свой': 2230, 'на': 2222, 'строительство': 1878, 'высокий': 1374, 'получение': 1314, 'коммерческий': 1311, 'для': 1289, 'жилой': 1230, 'сделка': 941, 'дополнительный': 906, 'заниматься': 904, 'подержать': 853, 'проведение': 768, 'сыграть': 765, 'сдача': 651, 'семья': 638, 'собственный': 635, 'со': 627, 'ремонт': 607, 'приобретение': 461, 'профильный': 436, 'подержанный': 111, '\n': 1})


**Вывод**

На данным этапе получили уникальные значения purpose, склеили их в одну строку с помощью join и лемматизировали. С помощь Counter посчитали частоту появления слов. Проанализировав частоту появления слов, выделим основные группы: - недвижимость (если встречаются слова жилье или недвижимость), - автомобиль (автомобиль), - образование (образование), - свадьба (свадьба).

### Категоризация данных

В дальнейшем перед нами 4 задачи - это проанализовать влияние наличия детей, семейного положения, уровня дохода и цели кредита на погашение кредита в срок. Для этого, проверим и выделим основные категории для каждого из этих параметров.

Начнем с цели кредита. Мы выделили 4 основные категории, теперь проверим количество клиентов по каждой из цели. Для этого напишем функцию lemmas_purpose:

In [20]:
# написание функции lemmas_purpose, которая возвращает значение в зависимости от цели
def lemmas_purpose(row):
    purpose = row['purpose']
    purpose_lemmas = m.lemmatize(purpose)
    if ('недвижимость' in purpose_lemmas or 'жилье' in purpose_lemmas):
        return 'жилье'
    elif 'автомобиль'in purpose_lemmas:
        return 'автомобиль'
    elif 'образование' in purpose_lemmas:
        return 'образование'
    elif 'свадьба' in purpose_lemmas:
        return 'свадьба'
    else:
        return 'цель не определена'
# создание нового столбца после применения функции lemmas_purpose к столбцу purpose и выявление частоты значений
data['purpose_group'] = data.apply(lemmas_purpose, axis = 1)
data['purpose_group'].value_counts()

жилье          10811
автомобиль      4306
образование     4013
свадьба         2324
Name: purpose_group, dtype: int64

Видно, что группы выделены успешно и все цели определены. В дальнейшем сможем проанализировать влияние цели кредита на возврат в срок.

Затем проверим какие уникальные значения есть в столбце family_status, удалив все дубликаты.

In [21]:
# cоздание новой таблицы family_category
# удаление дубликатов в таблице family_category и распечатка её в порядке возрастания значений столбца family_status_id
family_category = data[['family_status_id','family_status']]
family_category.drop_duplicates().reset_index(drop=True).sort_values('family_status_id',ascending=True)

family_status_id          family_status
0                 0        женат / замужем
1                 1       гражданский брак
2                 2         вдовец / вдова
3                 3              в разводе
4                 4  Не женат / не замужем

Видно, что в столбце family_status 5 уникальных статусов, их и будем использовать для группировки.

Для столбца children нужно два значения: есть дети и нет детей. Создадим функцию having_children, которая будет возвращать эти значения:

In [22]:
# написание функции having_children, которая возвращает значение в зависимости от цели
def having_children(row):
    children = row['children']
    if children <= 0:
        return 'нет детей'
    else:
        return 'есть дети'
# создание нового столбца после применения функции having_children к столбцу children и выявление частоты значений
data['children_group'] = data.apply(having_children, axis = 1)
data['children_group'].value_counts()

нет детей    14138
есть дети     7316
Name: children_group, dtype: int64

Таким образом, было выделено 2 группы клиентов. Без детей в 2 раза больше клиентов, в дальнейшем проверим, как это влияет на возврат кредита в срок.

Осталось выделить категории в столбце total_income, то есть уровень дохода. Для начала посмотрим максимальное и минимальное значение этого столбца:

In [23]:
# вывод минимального и максимального дохода
print('Минимальный доход клиента:', min(data['total_income']))
print('Максимальный доход клиента:', max(data['total_income']))

Минимальный доход клиента: 20667
Максимальный доход клиента: 2265604


Разброс в доходах достаточно большой. Разделим весь диапазон на 6 классов в зависимости от дохода:

In [24]:
# написание функции how_much, которая возвращает значение в зависимости от цели
def how_much(row):
    total_income = row['total_income']
    if total_income <= 100000:
        return 'нижний средний класс'
    elif 100000 < total_income <= 150000:
        return 'предсредний класс'
    elif 150000 < total_income <= 250000:
        return 'средний класс'
    elif 250000 < total_income <= 500000:
        return 'верхний средний класс'
    elif 500000 < total_income <= 1000000:
        return 'состоятельный'
    elif total_income > 1000000:
        return 'богатый'
# создание нового столбца после применения функции how_much к столбцу total_income и выявление частоты значений
data['total_income_group'] = data.apply(how_much, axis = 1)
data['total_income_group'].value_counts()

предсредний класс        7160
средний класс            7018
нижний средний класс     4463
верхний средний класс    2591
состоятельный             197
богатый                    25
Name: total_income_group, dtype: int64

Можем наблюдать, что большинство людей в предсреднем и среднем классах, что не так уж и плохо. Но богатых и состоятельных можно посчитать на "пальцах". Чуть дальше проверим, как доход влияет на возврат кредита в срок.

Проверим, все ли категории появились в таблице, чтобы можно было продолжить работу с данными и выявить, кто же возвращает кредит в срок чаще. Для этого выведем 10 строк нашей таблицы:

In [25]:
data.head(10)

children  days_employed  dob_years education  education_id  \
0         1           8437         42    высшее             0   
1         1           4024         36   среднее             1   
2         0           5623         33   среднее             1   
3         3           4124         32   среднее             1   
4         0          14177         53   среднее             1   
5         0            926         27    высшее             0   
6         0           2879         43    высшее             0   
7         0            152         50   среднее             1   
8         2           6929         35    высшее             0   
9         0           2188         41   среднее             1   

      family_status  family_status_id gender income_type  debt  total_income  \
0   женат / замужем                 0      F   сотрудник     0        253875   
1   женат / замужем                 0      F   сотрудник     0        112080   
2   женат / замужем                 0      M   сотрудник     0        145885   
3   женат / замужем                 0      M   сотрудник     0        267628   
4  гражданский брак                 1      F   пенсионер     0        158616   
5  гражданский брак                 1      M   компаньон     0        255763   
6   женат / замужем                 0      F   компаньон     0        240525   
7   женат / замужем                 0      M   сотрудник     0        135823   
8  гражданский брак                 1      F   сотрудник     0         95856   
9   женат / замужем                 0      M   сотрудник     0        144425   

                      purpose purpose_group children_group  \
0               покупка жилья         жилье      есть дети   
1     приобретение автомобиля    автомобиль      есть дети   
2               покупка жилья         жилье      нет детей   
3  дополнительное образование   образование      есть дети   
4             сыграть свадьбу       свадьба      нет детей   
5               покупка жилья         жилье      нет детей   
6           операции с жильем         жилье      нет детей   
7                 образование   образование      нет детей   
8       на проведение свадьбы       свадьба      есть дети   
9     покупка жилья для семьи         жилье      нет детей   

      total_income_group  
0  верхний средний класс  
1      предсредний класс  
2      предсредний класс  
3  верхний средний класс  
4          средний класс  
5  верхний средний класс  
6          средний класс  
7      предсредний класс  
8   нижний средний класс  
9      предсредний класс

**Вывод**

Отлично, все клиенты классифицированы и их можно группировать по интересующим нам параметрам при помощи сводных таблиц. Таким образом, у нас в зависимости: от цели кредита - 4 группы, от семейного положения - 5 групп, от наличия детей - 2 группы и от суммы дохода - 6 групп.

## Результаты анализа

### Есть ли зависимость между наличием детей и возвратом кредита в срок?

Первый вопрос влияет ли наличие детей на возврат кредита в срок. Создадим сводную таблицу, которая посчитает сколько людей имеют задолженность в зависимости от наличия детей. Значение 1 в столбце debt - имеется задолженность, 0 - отсутствует:

In [26]:
# создание сводной таблицы о количестве должников в зависимости от наличия детей
report_children = data.pivot_table(index = ['children_group'], values = 'debt', aggfunc = ['sum', 'count', 'mean'])
report_children.columns = ['debt', 'total', '%']
report_children = report_children.sort_values(by = ['total'], ascending = False)
report_children

debt  total         %
children_group                       
нет детей       1064  14138  0.075258
есть дети        677   7316  0.092537

**Вывод**

Столбец dept показывает количество клиентов, которые имеют задолженность, total - общее количетсво клиентов, % - доля должников. Из таблицы можно сказать, что клиентов без детей в два раза больше, чем с детьми и они чаще возвращают кредит в срок. Всего должников с детьми порядка 9%, а без детей - 7,5%. Таким образом, клиенты без детей вероятнее погасят кредит в срок, это может быть связано с тем, что у них нет дополнительных расходов на детей.

### Есть ли зависимость между семейным положением и возвратом кредита в срок?

По аналогии с предыдущим вопросом, создадим сводную таблицу, которая посчитает сколько людей имеют задолженность в зависимости от семейного положения:

In [27]:
# создание сводной таблицы о количестве должников в зависимости от семейного положения
report_family = data.pivot_table(index = ['family_status'], values = 'debt', aggfunc = ['sum', 'count', 'mean'])
report_family.columns = ['debt', 'total', '%']
report_family = report_family.sort_values(by = ['total'], ascending = False)
report_family

debt  total         %
family_status                               
женат / замужем         931  12339  0.075452
гражданский брак        388   4151  0.093471
Не женат / не замужем   274   2810  0.097509
в разводе                85   1195  0.071130
вдовец / вдова           63    959  0.065693

**Вывод**

Видно, что наибольшее количество должников не женаты / не замужем (порядка 10%) и в гражданском браке (9,3%). Максимально ответственно к возврату кредита относятся вдовцы / вдовы (всего 6,5%). Анализируя данную выборку, можно сделать вывод, что кредит целесообразно давать женатым / замужним, так как их большинство и не высок процент должников (7,5%). Это может быть связано с тем, что на семью приходится 2 зарплаты (если взять идеальную ситуацию).

### Есть ли зависимость между уровнем дохода и возвратом кредита в срок?

Уже отточенным методом из прошлых вопросов, создадим сводную таблицу, которая посчитает сколько людей имеют задолженность в зависимости от уровня дохода:

In [28]:
# создание сводной таблицы о количестве должников в зависимости от уровня дохода
report_income = data.pivot_table(index = ['total_income_group'], values = 'debt', aggfunc = ['sum', 'count', 'mean'])
report_income.columns = ['debt', 'total', '%']
report_income = report_income.sort_values(by = ['total'], ascending = False)
report_income

debt  total         %
total_income_group                          
предсредний класс       624   7160  0.087151
средний класс           569   7018  0.081077
нижний средний класс    354   4463  0.079319
верхний средний класс   180   2591  0.069471
состоятельный            12    197  0.060914
богатый                   2     25  0.080000

**Вывод**

Посмотрев на полученную таблицу, можно сделать вывод, что зависимость от дохода менее явная, чем от семейного положения или наличия детей. На удивление, состоятельные и богатые тоже не всегда платят вовремя. Основные группы клиентов это нижний средний, предсредний и средний класс. Процент должников у эти групп примерно одинаков: 8 - 8,5%. Состоятельных людей намного меньше, но они платят исправней (всего 6% должников). В целом, большинство клиентов возвращает кредит вовремя, хочется верить, что люди берут кредит, исходя из своих возможностей.

### Как разные цели кредита влияют на его возврат в срок?

И наконец последний фактов, влияние которого нужно проверить. Строим уже известную нам сводную таблицу, которая посчитает сколько людей имеют задолженность в зависимости от цели кредита:

In [29]:
# создание сводной таблицы о количестве должников в зависимости от цели кредита
report_purpose = data.pivot_table(index = ['purpose_group'], values = 'debt', aggfunc = ['sum', 'count', 'mean'])
report_purpose.columns = ['debt', 'total', '%']
report_purpose = report_purpose.sort_values(by = ['total'], ascending = False)
report_purpose

debt  total         %
purpose_group                       
жилье           782  10811  0.072334
автомобиль      403   4306  0.093590
образование     370   4013  0.092200
свадьба         186   2324  0.080034

**Вывод**

Большинство людей берут кредит на жилье (свое гнездышко - здорово) и очень ответственно платят по нему (всего 7,2 % должников). Больше всего должников у тех, кто берет кредит на автомобиль или образование (примерно 9,3% для каждых). Это может быть связано с тем, что автомобиль требует дополнительных вложений (ремонт, топливо), а образование не дает работать на полную ставку (больше инвестиция на будущее).

## Вывод

Каждый фактор влияет на возврат кредита в срок. Проанализировав все зависимости, можно сделать вывод, что самое большое влияние на возврат кредита - это семейное положение. Вне зависимости от дохода, количество должников не сильно разнится. 

На мой взгляд, общий процент должников не так уж и высок, вне зависимости от любых факторов он не превышает 10%. Если составить портрет идеального заемщика, то человек должен быть женатый / замужней, в этой семье нет детей, кредит берут на жилье и доход чуть выше среднего.